### PR TITLE
hooks: substitute env vars when executing hooks

### PR DESCRIPTION
**Pull request by stolowski**
This fixes the problem reported here https://forum.snapcraft.io/t/declaratively-defining-environment-variables/175/13

snap-exec lacked Substitute.Env(..) for hooks, likely due to an oversight; we do that for apps.

### CODE REVIEW

File: tests/main/hooks-env/task.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+summary: Check that environment variables are interpolated when running hooks.
+
+prepare: |
+    snapbuild $TESTSLIB/snaps/snapctl-hooks .
+    snap install --dangerous snapctl-hooks_1.0_all.snap
+
+restore: |
+    rm snapctl-hooks_1.0_all.snap
```
Comment (zyga) [inline tests/main/hooks-env/task.yaml]: You probably don't need this if you use one of the snap build helpers. They also cache the build across tests.

Comment (stolowski) [reply tests/main/hooks-env/task.yaml]: Good point, I keep forgetting about the build helper. Done.

File: tests/main/hooks-env/task.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+environment:
+    ENVDUMP: /var/snap/snapctl-hooks/current/hooks-env
+
+execute: |
+    snap set snapctl-hooks command=dump-env
+
+    echo "Test that environment variables were interpolated"
+    cat $ENVDUMP | MATCH "^TEST_COMMON=/var/snap/snapctl-hooks/common$"
+    cat $ENVDUMP | MATCH "^TEST_DATA=/var/snap/snapctl-hooks/.*$"
+    cat $ENVDUMP | MATCH "^TEST_SNAP=/snap/snapctl-hooks/.*$"
```
Comment (niemeyer) [inline tests/main/hooks-env/task.yaml]: Test looks nice, but can we please move this into snap-run-hook which is where we already test the basic hook running logic?

**Comment by stolowski** [comment]
Ok, I made the changes to spread tests.

**Pull request merged**